lib/src/modules/command.cc

```diff
@@ -0,0 +1,71 @@
+#include <pxp-agent/modules/command.hpp>
+#include <pxp-agent/configuration.hpp>
+#include <leatherman/file_util/file.hpp>
+#include <boost/filesystem/path.hpp>
+#include <boost/algorithm/string.hpp>
+#include <boost/tokenizer.hpp>
+
+namespace PXPAgent {
+namespace Modules {
+
+namespace fs = boost::filesystem;
+namespace lth_jc = leatherman::json_container;
+namespace lth_file = leatherman::file_util;
+
+static const std::string COMMAND_RUN_ACTION { "run" };
+
+Command::Command(const fs::path& exec_prefix, std::shared_ptr<ResultsStorage> storage) :
+    BoltModule { exec_prefix, std::move(storage) }
+{
+    module_name = "command";
+    actions.push_back(COMMAND_RUN_ACTION);
+
+    // Command actions require a single "command" string parameter
+    PCPClient::Schema input_schema { COMMAND_RUN_ACTION };
+    input_schema.addConstraint("command", PCPClient::TypeConstraint::String, true);
+    input_validator_.registerSchema(input_schema);
+
+    PCPClient::Schema output_schema { COMMAND_RUN_ACTION };
+    results_validator_.registerSchema(output_schema);
+}
+
+Util::CommandObject Command::buildCommandObject(const ActionRequest& request)
+{
+    const auto params = request.params();
+
+    assert(params.includes("command") &&
+           params.type("command") == lth_jc::DataType::String);
+
+    const fs::path& results_dir = request.resultsDir();
+
+    // Try to split the raw command text into executable and arguments, as leatherman::execution
+    // expects, taking care to retain spaces within quoted executables or argument names:
+    auto raw_command = params.get<std::string>("command");
```

Comment:
Looks like this was pulled from program_options::split_unix. Might be worth calling out the origin.